Framework/PythonInterface/plugins/algorithms/DPDFreduction.py

```diff
@@ -131,50 +131,51 @@ def PyExec(self):
         if self._clean:
             prefix = '__'
         # "wn" denotes workspace name
-        wn_data = prefix + 'data'
-        wn_van = prefix + 'vanadium'
-        wn_reduced = prefix + 'reduced'
-        wn_ste = prefix + 'S_theta_E'
+        wn_data = prefix + 'data'  # Accumulated data events
+        wn_data_mon = prefix + 'data_monitors'  # Accumulated monitors for data
+        wn_van = prefix + 'vanadium'  # White-beam vanadium
         wn_van_st = prefix + 'vanadium_S_theta'
+        wn_reduced = prefix + 'reduced'  # data after DGSReduction
+        wn_ste = prefix + 'S_theta_E'  # data after grouping by theta angle
         wn_sten = prefix + 'S_theta_E_normalized'
-        wn_steni = prefix + 'S_theta_E_normalized_interp'
+        wn_steni = prefix + 'S_theta_E_interp'
         wn_sqe = prefix + 'S_Q_E'
         wn_sqeb = prefix + 'S_Q_E_binned'
         wn_sqesn = prefix + wn_sqes + '_norm'
         # Empty can files
-        wn_ec_data = prefix + 'ec_data'
-        wn_ec_reduced = prefix + 'ec_reduced'
-        wn_ec_ste = prefix + 'ec_S_theta_E'
+        wn_ec_data = prefix + 'ec_data'  # Accumulated empty can data
+        wn_ec_data_mon = prefix + 'ec_data_monitors'  # Accumulated monitors for empty can
+        wn_ec_reduced = prefix + 'ec_reduced'  # empty can data after DGSReduction
+        wn_ec_ste = prefix + 'ec_S_theta_E'  # empty can data after grouping by theta angle
+
 
         datasearch = config["datasearch.searcharchive"]
         if datasearch != "On":
             config["datasearch.searcharchive"] = "On"
 
         try:
-            # Load several event files into a single workspace. The nominal incident
-            # energy should be the same to avoid difference in energy resolution
-            api.Load(Filename=self._runs, OutputWorkspace=wn_data)
-
             # Load the vanadium file, assumed to be preprocessed, meaning that
             # for every detector all events within a particular wide wavelength
             # range have been rebinned into a single histogram
             api.Load(Filename=self._vanfile, OutputWorkspace=wn_van)
+            # Check for white-beam vanadium, true if the vertical chopper is absent (vChTrans==2)
+            if mtd[wn_van].run().getProperty('vChTrans').value[0] != 2:
```

Comment:
This works only on ARCS and SEQ. At some point in the code you MUST check if the instrument is ARCS. I would do it as part of the `_load` function. Just raise a `RuntimeError` or a `ValueError` or a `NotImplementedError`